src/pkg/packaging-tools/framework.dependency.targets

```diff
@@ -251,7 +252,17 @@
     </PropertyGroup>
 
     <PropertyGroup Condition="'@(_runtimeJIT)' != ''">
-      <_jitPath>%(_runtimeJIT.FullPath)</_jitPath>
+      <_runtimeJITFilterDir>$([System.IO.Path]::DirectorySeparatorChar)$(PackageRID)$([System.IO.Path]::DirectorySeparatorChar)</_runtimeJITFilterDir>
+    </PropertyGroup>
+
+    <!-- It is possible for _runtimeJIT to pick up 2 jit files:
+         one for cross compilation and one for native compilation. -->
+    <ItemGroup Condition="'@(_runtimeJIT)' != '' AND '$(_crossDir)' == ''">
+      <_filteredRuntimeJIT Include="%(_runtimeJIT.Identity)" Condition="'@(_runtimeJIT->Contains($(_runtimeJITFilterDir)))' == 'True'" />
```

Comment:
This seems ok, but I think it would be more readable to change how `_runtimeJIT` is discovered:

```xml
<_runtimeJIT Include="$(_jitPackageDir)**/$(LibraryFilePrefix)clrjit$(LibraryFileExtension)" />
```

to make it more specific, maybe:

```xml
<!-- Find cross-arch JIT if specified, otherwise use current PackageRID. -->
<_runtimeJIT Include="$(_jitPackageDir)runtimes$(_crossDir)/native/$(LibraryFilePrefix)clrjit$(LibraryFileExtension)" Condition="'$(_crossDir)' != ''"/>
<_runtimeJIT Include="$(_jitPackageDir)runtimes/$(PackageRID)/native/$(LibraryFilePrefix)clrjit$(LibraryFileExtension)" Condition="'$(_crossDir)' == ''" />
```

I might be missing some subtlety here.

(I don't *think* that losing the flexibility of `**` is in itself a problem. That flexibility seems more likely to me to be a bad thing, e.g. if CoreCLR started providing an arm64 => x64 cross-JIT, the x64 => x64 build could get confused with our current code (before your PR).)